torch/csrc/jit/script/init.cpp

```diff
@@ -852,19 +853,38 @@ void initJitScriptBindings(PyObject* module) {
         cu->register_type(classType);
         std::vector<ResolverPtr> rcbs;
         std::vector<Def> methodDefs;
-        for (const auto& def : classDef.body()) {
-          if (def.kind() != TK_DEF) {
-            throw ErrorReport(def.range())
+        std::vector<std::pair<Def, FunctionDefaults>> all_defaults;
```

Comment:
What is preventing this reusing the existing code path?

Reply:
Classes, functions, and methods all access the compiler in their own way, and the defaults are installed on the schema after the fact to avoid complicating the compiler

Reply:
"to avoid complicating the compiler" we introduced 4 different paths lol. although i know a number of different pathways rely on this so it'd be hard to change (e.g. overloads).